app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt

```diff
@@ -121,13 +137,40 @@ class TabTrayDialogFragment : AppCompatDialogFragment(), TabTrayInteractor {
     }
 
     override fun onSaveToCollectionClicked() {
+        // Only register the observer right before moving to collection creation
+        registerCollectionStorageObserver()
         interactor?.onSaveToCollectionClicked()
     }
 
     override fun onCloseAllTabsClicked(private: Boolean) {
         interactor?.onCloseAllTabsClicked(private)
     }
 
+    private fun registerCollectionStorageObserver() {
+        requireComponents.core.tabCollectionStorage.register(collectionStorageObserver, this)
+    }
+
+    private fun showCollectionSnackbar() {
+        view.let {
+            val snackbar = FenixSnackbar
+                .make(
+                    duration = FenixSnackbar.LENGTH_LONG,
+                    isDisplayedWithBrowserToolbar = true,
+                    view = (view as View)
+                )
+                .setText(requireContext().getString(R.string.create_collection_tabs_saved))
+                .setAction(requireContext().getString(R.string.create_collection_view)) {
+                    dismissAllowingStateLoss()
+                    
+                    val directions = TabTrayDialogFragmentDirections.actionGlobalHome()
```

Comment:
This navigation works when on Home but doesn't work in Browser fragment.  Any ideas @boek ?